.github/workflows/publish.yml

```diff
@@ -6,20 +6,57 @@ jobs:
   main:
     name: Publish
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
 
     steps:
-      - uses: actions/checkout@v1
+      - name: Checkout repo
+        uses: actions/checkout@v3
 
       - name: Use Node.js 18.x
         uses: actions/setup-node@v2.2.0
         with:
-          node-version: '18.x'
+          node-version: "18.x"
 
       - name: Install
         run: yarn
 
-      - name: format
+      - name: Format
         run: yarn format
 
       - name: Check formatting
-        run: git diff --exit-code
\ No newline at end of file
+        run: git diff --exit-code
+
+      # TODO: Remove and modify "Create Catalog" once argo-tasks released
+      - name: Get Collections
+        run: |
+          echo "$(find stac/ -name collection.json)" > files.txt
+
+      # catalog.json is not pushed to the repository (temporary solution)
+      - name: Create Catalog
+        uses: docker://ghcr.io/linz/argo-tasks:v2
+        with:
+          args: stac-catalog --output stac/catalog.json --template template/catalog.json --collections files.txt
+
+      - name: Validate STAC
+        run: |
+          # Enable double star operator
+          shopt -s globstar
+          docker run -v $PWD:$PWD ghcr.io/linz/argo-tasks:v2 stac-validate $PWD/stac/**/collection.json
+
+      # FIXME: issue with the latest version (v2.0.0): https://github.com/aws-actions/configure-aws-credentials/issues/680
```

Comment:
That is a deprecation warning? What affect does it have on using t he master version?

Reply:
Weirdly I had an issue while using `v2` which was actually related to the `permission` that I haven't set. So not related to that at all... I'll upgrade.